src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                                            bundle.getTeamNameForEmail(participantIdentifier)));
         } else {
-            result = "<a class=\"btn btn-primary btn-block btn-tm-actions session-publish-for-test" 
-                   + (hasPublish ? "\"" : DISABLED) + "href=\""
-                   + getInstructorFeedbackPublishLink(session.courseId, session.feedbackSessionName, isHome) 
-                   + "\" " + "title=\""
-                   + (hasPublish ? Const.Tooltips.FEEDBACK_SESSION_PUBLISH 
-                                 : Const.Tooltips.FEEDBACK_SESSION_AWAITING)
-                   + "\" " + "data-toggle=\"tooltip\" data-placement=\"top\""
-                   + (hasPublish ? "onclick=\"return togglePublishEvaluation('" + session.feedbackSessionName + "', " 
-                                                                                + session.isPublishedEmailEnabled + ");\" " 
-                                              : " ") 
-                   + disablePublishSessionStr + ">Publish Results</a> ";
-        }
-        return result;
-    }
-
-    public String getResultsVisibleFromText() {
-        if (bundle.feedbackSession.resultsVisibleFromTime.equals(Const.TIME_REPRESENTS_FOLLOW_VISIBLE)) {
-            if (bundle.feedbackSession.sessionVisibleFromTime.equals(Const.TIME_REPRESENTS_FOLLOW_OPENING)) {
-                return TimeHelper.formatTime(bundle.feedbackSession.startTime);
-            } else if (bundle.feedbackSession.sessionVisibleFromTime.equals(Const.TIME_REPRESENTS_NEVER)) {
-                return "Never";
-            } else {
-                return TimeHelper.formatTime(bundle.feedbackSession.sessionVisibleFromTime);
+            responseRows.addAll(buildMissingResponseRowsBetweenRecipientAndPossibleGivers(
+                                            question, possibleParticipantsWithoutResponses, participantIdentifier, 
+                                            bundle.getNameForEmail(participantIdentifier),
+                                            bundle.getTeamNameForEmail(participantIdentifier)));
+        }
+        
+        
+        return responseRows;
+    }
+    
+
+    private void configureResponseRow(FeedbackQuestionAttributes question,
+                                                 String giver, String recipient,
+                                                 InstructorResultsResponseRow responseRow) {
+        
+        switch (viewType) {
+            case QUESTION:
+                responseRow.setGiverProfilePictureDisplayed(validator.getInvalidityInfo(FieldValidator.FieldType.EMAIL, giver).isEmpty());
+                responseRow.setGiverProfilePictureLink(new Url(getProfilePictureLink(giver)));
+                
+                responseRow.setRecipientProfilePictureDisplayed(validator.getInvalidityInfo(FieldValidator.FieldType.EMAIL, recipient).isEmpty());
+                responseRow.setRecipientProfilePictureLink(new Url(getProfilePictureLink(recipient)));
```

Comment:
Just wondering, but would it be more suitable to have a `null` value for profile picture link represent "profile picture not displayed"?
I can't think of any reason to set a profile picture link, then not display it.